src/Interpreters/DatabaseCatalog.cpp

```diff
@@ -344,7 +344,8 @@ DatabaseAndTable DatabaseCatalog::getTableImpl(
     DatabasePtr database;
     {
         std::lock_guard lock{databases_mutex};
-        auto it = databases.find(table_id.getDatabaseName());
+        // Callers assume that this method doesn't throw exceptions, but getDatabaseName() will throw if there is no database part.
+        auto it = table_id.hasDatabase() ? databases.find(table_id.getDatabaseName()) : databases.end();
```

Comment:
The only downside I see here is a bit cryptic error message in the `system.errors`: something like "Database doesn't exists".

Slightly better alternative would be to generate different error message, something like "Database is empty".
But that still would be a very confusing to the end user since effectively there is no error, and calling code handles this situation correctly.

We can't just return the empty object without creating `Exception` object, since that would break `DatabaseCatalog::getTable`.

I considered adding special case into `tryGetTable`, and `tryGetDatabaseAndTable`, to handle empty database name. But that looks worse from a maintenance perspective: somebody is deemed to forged about this special case down the road and most likely to re-introduce similar error.

So the choice here seems to be between two bogus error messages.